job-service-db/src/main/resources/db/migration/V3.0.0/V3__job_service.sql

```diff
@@ -0,0 +1,690 @@
+    ADD CONSTRAINT fk_job_task_data FOREIGN KEY (partition_id, job_id) REFERENCES job (partition_id, job_id);
+
+ALTER TABLE job_task_data
+    ADD CONSTRAINT pk_job_task_data PRIMARY KEY (partition_id, job_id);
+
+-------- public."label"
+
+CREATE TABLE IF NOT EXISTS public."label"
+(
+    partition_id varchar(40)  NOT NULL DEFAULT 'default'::character varying,
+    job_id       varchar(48)  NOT NULL,
+    "label"      varchar(255) NOT NULL,
+    value        varchar(255) NULL,
+    CONSTRAINT label_pkey PRIMARY KEY (partition_id, job_id, label),
+    CONSTRAINT fk_label_job FOREIGN KEY (partition_id, job_id) REFERENCES public.job (partition_id, job_id)
+);
+
+-------- public.stowed_task
+
+CREATE TABLE IF NOT EXISTS public.stowed_task
+(
+    partition_id                               varchar(40)  NOT NULL,
+    job_id                                     varchar(48)  NOT NULL,
+    task_classifier                            varchar(255) NOT NULL,
+    task_api_version                           int4         NOT NULL,
+    task_data                                  bytea        NOT NULL,
+    task_status                                varchar(255) NOT NULL,
+    context                                    bytea        NOT NULL,
+    "to"                                       varchar(255) NOT NULL,
+    tracking_info_job_task_id                  varchar(255) NOT NULL,
+    tracking_info_last_status_check_time       int8         NULL,
+    tracking_info_status_check_interval_millis int8         NULL,
+    tracking_info_status_check_url             text         NULL,
+    tracking_info_tracking_pipe                varchar(255) NULL,
+    tracking_info_track_to                     varchar(255) NULL,
+    source_info                                bytea        NULL,
+    correlation_id                             varchar(255) NULL,
+    CONSTRAINT fk_stowed_task FOREIGN KEY (partition_id, job_id) REFERENCES public.job (partition_id, job_id)
+);
+
+-- Drop Indexes
+DROP INDEX IF EXISTS idx_job_jobid_status;
+
+-- Create Indexes
+CREATE INDEX IF NOT EXISTS idx_deleted_parent_table_log
+    ON public.deleted_parent_table_log
+        USING btree (table_name);
+
+CREATE INDEX IF NOT EXISTS idx_delete_log_table_name
+    ON public.delete_log
+        USING btree (table_name);
+
+CREATE INDEX IF NOT EXISTS idx_job_create_date
+    ON public.job
+        USING btree (create_date);
+
+CREATE INDEX IF NOT EXISTS idx_partition_id_and_job_id
+    ON public.stowed_task
+        USING btree (partition_id, job_id);
+
+CREATE INDEX IF NOT EXISTS idx_job_partition_id_and_dependent_job_id
+    ON public.job_dependency
+        USING btree (partition_id, dependent_job_id);
+
+/*
+ *
+ * Drop unused functions and forward declare internal functions
+ *
+ */
+
+DROP FUNCTION IF EXISTS cancel_job(
+    in_partition_id VARCHAR(40),
+    in_job_id VARCHAR(48),
+    in_short_job_id VARCHAR(48)
+);
+CREATE OR REPLACE FUNCTION internal_cleanup_completed_subtask_report(
```

Comment:
Would it be tidier to do all the drops first and then to do the forward declarations rather than mixing them?  Maybe the drops go in a V4 and the forward declares in a V5?  (Of course I understand that going forward they will be mixed chronologically as changes take place but would it be tidier at this starting point?)

Reply:
Yes, unsure of the V3, V4, V5, maybe more  `V3__job_service_database_schema.sql` ,   `V3__job_service_functions.sql`, ... Let me know